apps/roam/src/components/settings/SuggestiveModeSettings.tsx

```diff
@@ -11,21 +11,30 @@ import { createOrUpdateDiscourseEmbedding } from "~/utils/syncDgNodesToSupabase"
 import { render as renderToast } from "roamjs-components/components/Toast";
 
 const SuggestiveModeSettings = () => {
-  const settings = getFormattedConfigTree();
+  const [settings, setSettings] = useState(() => getFormattedConfigTree());
 
   const [suggestiveModeUid, setSuggestiveModeUid] = useState(
     settings.suggestiveMode.parentUid,
   );
+  const [pageGroupsUid, setPageGroupsUid] = useState(
+    settings.suggestiveMode.pageGroups.uid,
+  );
   useEffect(() => {
-    if (suggestiveModeUid) return;
+    if (pageGroupsUid) return;
     void (async () => {
       const smUid = await createBlock({
         parentUid: getPageUidByPageTitle(DISCOURSE_CONFIG_PAGE_TITLE),
         node: { text: "Suggestive Mode" },
       });
+      const pgUid = await createBlock({
+        parentUid: smUid,
+        node: { text: "Page Groups" },
+      });
       setSuggestiveModeUid(smUid);
+      setPageGroupsUid(pgUid);
+      setSettings(getFormattedConfigTree());
```

Comment:
Ideally we grab the settings from the "db" once.  Then we manage then locally after that.

So in this case we would update the local `settings` variable directly when we call `setPageGroupsUid` and `setSuggestiveMode` uid